### PR TITLE
secboot: factor code for encrypted disk activation

### PR DESCRIPTION
**Pull request by valentindavid**
This factors and simplifies the TPM and FDE Hook code together. This does not yet factor the key file base one.

### CODE REVIEW

**Review by pedronis — COMMENTED** [review]
thanks, some small remarks and questions

File: secboot/secboot_sb.go [hunk 1]
```diff
@@ -120,13 +120,57 @@ func UnlockVolumeUsingSealedKeyIfEncrypted(disk disks.Disk, name string, sealedE
...
+	}
+	// TODO: set boot mode
+	//sbSetBootMode("run")
+	//defer sbSetBootMode("")
+	sbSetKeyRevealer(&keyRevealerV3{})
+	defer sbSetKeyRevealer(nil)
+
+	options := activateVolOpts(opts.AllowRecoveryKey)
+	// TODO: remove this
+	options.Model = sb.SkipSnapModelCheck
```
[the remark this answers is not in the document]
Comment (valentindavid) [reply secboot/secboot_sb.go:153]: Yes, I will have to remove it with the rebase on top of #14573.

File: secboot/secboot_sb_test.go [hunk 2]
```diff
@@ -699,20 +665,14 @@ func (s *secbootSuite) TestUnlockVolumeUsingSealedKeyIfEncrypted(c *C) {
 					Model:            sb.SkipSnapModelCheck,
 				})
 			}
-			if !tc.activated && tc.activateErr == nil {
-				return errors.New("activation error")
-			}
 			return tc.activateErr
```
[the remark this answers is not in the document]
Comment (valentindavid) [reply secboot/secboot_sb_test.go:668]: Yes. In test cases 0, 1, 3, 4, 6, 8, 9.

**Review by pedronis — APPROVED** [review]
thank you

File: secboot/secboot_sb.go
Comment (alfonsosanchezbeato) [inline secboot/secboot_sb.go:1]: nitpick: the changes here could be in a separate function as before the change

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks

File: secboot/secboot_sb.go
Comment (valentindavid) [reply secboot/secboot_sb.go:1]: I will add a TODO. We need to do more changes in there.

**Pull request merged**